code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm

```diff
@@ -410,19 +410,412 @@ datum/reagent/consumable/eggyolk
+
+datum/reagent/beans
+	name = "Refried beans"
+	id = "beans"
+	description = "A dish made of mashed beans cooked with lard."
+	reagent_state = LIQUID
+	color = "#684435"
+
+/datum/reagent/bread
+	name = "Bread"
+	id = "bread"
+	description = "Bread! Yep, bread."
+	reagent_state = SOLID
+	color = "#9C5013"
+
+/datum/reagent/vomit
+	name = "Vomit"
+	id = "vomit"
+	description = "Looks like someone lost their lunch. And then collected it. Yuck."
+	reagent_state = LIQUID
+	color = "#FFFF00"
+
+/datum/reagent/greenvomit
+	name = "Green vomit"
+	id = "green_vomit"
+	description = "Whoa, that can't be natural. That's horrible."
+	reagent_state = LIQUID
+	color = "#78FF74"
+
+/datum/reagent/ectoplasm
+	name = "Ectoplasm"
+	id = "ectoplasm"
+	description = "A bizarre gelatinous substance supposedly derived from ghosts."
+	reagent_state = LIQUID
+	color = "#8EAE7B"
+
+datum/reagent/ectoplasm/reaction_mob(var/mob/M, var/method=TOUCH, var/volume)
+	if(!istype(M, /mob/living))
+		return
+	if(method == INGEST)
+		var/spooky_eat = pick("Ugh, why did you eat that? Your mouth feels haunted. Haunted with bad flavors.", "Ugh, why did you eat that? It has the texture of ham aspic.  From the 1950s.  Left out in the sun.", "Ugh, why did you eat that? It tastes like a ghost fart.", "Ugh, why did you eat that? It tastes like flavor died.")
+		M << "<span class = 'warning'>[spooky_eat]</span>"
+
+/datum/reagent/ectoplasm/on_mob_life(var/mob/living/M as mob)
+	var/spooky_message = pick("You notice something moving out of the corner of your eye, but nothing is there...", "Your eyes twitch, you feel like something you can't see is here...", "You've got the heebie-jeebies.", "You feel uneasy.", "You shudder as if cold...", "You feel something gliding across your back...")
+	if(prob(5))
+		M << "<span class='warning'>[spooky_message]</span>"
+	..()
+	return
+
+/datum/reagent/soybeanoil
+	name = "Space-soybean oil "
+	id = "soybeanoil"
+	description = "An oil derived from extra-terrestrial soybeans."
+	reagent_state = LIQUID
+	color = "#B1B0B0"
+
+/datum/reagent/soybeanoil/on_mob_life(var/mob/living/M as mob)
+	if(prob(5))
+		M.reagents.add_reagent("porktonium",5)
+	..()
+	return
+
+/datum/reagent/hydrogenated_soybeanoil
+	name = "Partially hydrogenated space-soybean oil"
+	id = "hydrogenated_soybeanoil"
+	description = "An oil derived from extra-terrestrial soybeans, with additional hydrogen atoms added to convert it into a saturated form."
+	reagent_state = LIQUID
+	color = "#B1B0B0"
+
+/datum/reagent/hydrogenated_soybeanoil/on_mob_life(var/mob/living/M as mob)
+	if(prob(8))
+		M.reagents.add_reagent("porktonium",5)
 	..()
+	return
+
+/datum/chemical_reaction/hydrogenated_soybeanoil
+	name = "Partially hydrogenated space-soybean oil"
+	id = "hydrogenated_soybeanoil"
+	result = "hydrogenated_soybeanoil"
+	required_reagents = list("soybeanoil" = 1, "hydrogen" = 1)
+	result_amount = 2
+	required_temp = 520
+	mix_message = "The mixture emits a burnt, oily smell."
+
```

Comment:
Three different chems, all just slightly different types of soy oil. Look at them features